### PR TITLE
Add TFTransformer example to README

### PR DESCRIPTION
**Pull request by yogeshg**
* add TFTransformer description and example
* restructure README for tensor input transformer examples
* update link to release-specific databricks notebooks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
 
-    The difference in the API from `TFImageTransformer` above stems from the fact that usual Keras workflows have very specific ways to load and resize images that are not part of the TensorFlow Graph.
+```python
+from keras.applications import InceptionV3
 
-    To use the transformer, we first need to have a Keras model stored as a file. For this example we'll just save the Keras built-in InceptionV3 model instead of training one.
+model = InceptionV3(weights="imagenet")
+model.save('/tmp/model-full.h5')  # saves to the local filesystem
+# move to a permanent place for future use
+dbfs_model_path = 'dbfs:/models/model-full.h5'
```
Comment (sueann) [inline README.md]: we don't want the dbfs stuff here because this README is for general spark usage. let's revert this cell to the previous version in this README (okay in the notebook). this is precisely the kind of the difference we have between the README and databricks notebooks.

File: README.md [hunk 2]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
+  image = np.expand_dims(image, axis=0)
+  return preprocess_input(image)
+
+dbutils.fs.cp(dbfs_model_path, 'file:/tmp/model-full-tmp.h5')
+transformer = KerasImageFileTransformer(inputCol="uri", outputCol="predictions",
+                                        modelFile='/tmp/model-full-tmp.h5',  # local file path for model
+                                        imageLoader=loadAndPreprocessKerasInceptionV3,
+                                        outputMode="vector")
+
+files = ["/dbfs" + str(f.path)[5:] for f in dbutils.fs.ls(sample_img_dir)]  # make "local" file paths for images
```
Comment (sueann) [inline README.md]: revert to the non-dbfs version

File: README.md [hunk 3]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
-    import os
-    from sparkdl import KerasImageFileTransformer
 
-    def loadAndPreprocessKerasInceptionV3(uri):
-        # this is a typical way to load and prep images in keras
-        image = img_to_array(load_img(uri, target_size=(299, 299)))
-        image = np.expand_dims(image, axis=0)
-        return preprocess_input(image)
+```python
+display(keras_pred_df.select("uri", "predictions"))
```
Comment (sueann) [inline README.md]: display is also a databricks notebook thing, so remove from here

File: README.md [hunk 4]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
-    predictions_df = predictor.transform(image_df)
-    ```
+image_df = readImages(sample_img_dir)
 
-2.  For TensorFlow users
+predictor = DeepImagePredictor(inputCol="image", outputCol="predicted_labels", modelName="InceptionV3", decodePredictions=True, topK=10)
+predictions_df = predictor.transform(image_df)
 
-    Deep Learning Pipelines provides a Transformer that will apply the given TensorFlow Graph to a DataFrame containing a column of images (e.g. loaded using the utilities described in the previous section). Here is a very simple example of how a TensorFlow Graph can be used with the Transformer. In practice, the TensorFlow Graph will likely be restored from files before calling `TFImageTransformer`.
+display(predictions_df.select("filePath", "predicted_labels"))
```
Comment (sueann) [inline README.md]: remove display which is in databricks notebooks only

File: README.md [hunk 5]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
+  return Row(inputCol = xy, label = l)
 
-### Deploying models as SQL functions
+samples_0 = [np.random.randn(2) + center_0 for _ in range(n_sample//2)]
+labels_0 = [0 for _ in range(n_sample//2)]
+samples_1 = [np.random.randn(2) + center_1 for _ in range(n_sample//2)]
+labels_1 = [1 for _ in range(n_sample//2)]
 
-One way to productionize a model is to deploy it as a [Spark SQL User Defined Function](https://docs.databricks.com/spark/latest/spark-sql/udf-in-python.html), which allows anyone who knows SQL to use it. Deep Learning Pipelines provides mechanisms to take a deep learning model and register a Spark SQL User Defined Function (UDF).
+# Make dataframe for the Spark use case
```
Comment (sueann) [inline README.md]: comment is confusing and unnecessary given the description above so remove

File: README.md [hunk 6]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
+* TensorFlow graphs
+* Keras models
 
-    files = [os.path.abspath(os.path.join(dirpath, f)) for f in os.listdir("/data/myimages") if f.endswith('.jpg')]
-    uri_df = sqlContext.createDataFrame(files, StringType()).toDF("uri")
+##### For TensorFlow users
 
-    final_df = transformer.transform(uri_df)
-    ```
+```python
```
Comment (sueann) [inline README.md]: better to put imports in the cell they are used in, since people are likely to try out copying a cell at a time

File: README.md [hunk 7]
```diff
@@ -147,160 +154,237 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
+from pyspark.sql.types import StringType
+from sparkdl import KerasImageFileTransformer
+
+def loadAndPreprocessKerasInceptionV3(uri):
+  # this is a typical way to load and prep images in keras
+  image = img_to_array(load_img(uri, target_size=(299, 299)))  # image dimensions for InceptionV3
+  image = np.expand_dims(image, axis=0)
+  return preprocess_input(image)
+
+dbutils.fs.cp(dbfs_model_path, 'file:/tmp/model-full-tmp.h5')
```
Comment (sueann) [inline README.md]: use non-dbfs version

File: README.md [hunk 8]
```diff
@@ -147,160 +157,226 @@ print("Training set accuracy = " + str(evaluator.evaluate(predictionAndLabels)))
...
-        image = img_to_array(load_img(uri, target_size=(299, 299)))
-        image = np.expand_dims(image, axis=0)
-        return preprocess_input(image)
+##### For TensorFlow users
 
-    transformer = KerasImageFileTransformer(inputCol="uri", outputCol="predictions",
-                                            modelFile="/tmp/model-full.h5",
-                                            imageLoader=loadAndPreprocessKerasInceptionV3,
-                                            outputMode="vector")
+First we generate sample dataset of 2-dimensional points, Gaussian distributed around two different centers
```
Comment (sueann) [inline README.md]: Before diving into an example, let's add a description for TFTransformer to give more information, like:

`TFTransformer` applies a user-specified TensorFlow graph to tensor inputs of up to 2 dimensions. The TensorFlow graph may be specified as TensorFlow graph objects (`tf.Graph` or `tf.GraphDef`) or checkpoint or `SavedModel` objects (see [the input object class](https://github.com/databricks/spark-deep-learning/blob/master/python/sparkdl/graph/input.py#L27) for more detail). The `transform()` function applies the TensorFlow graph to a column of arrays (where an array corresponds to a Tensor) in the input DataFrame and outputs a column of arrays corresponding to the output of the graph.
